pkg/clients/unstructured.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2016-2019, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clients
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/pulumi/pulumi-kubernetes/pkg/kinds"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func FromUnstructured(obj *unstructured.Unstructured) (metav1.Object, error) {
+	b, err := obj.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+
+	var output metav1.Object
+	switch kinds.Kind(obj.GetKind()) {
```

Comment:
Can we have docs stating which objects are supported today vs not?

Reply:
Yeah, I'll add docs eventually, but for right now, this is the set of objects we're using the in complex await logic.